XiEditor/Document.swift

```diff
@@ -176,9 +178,18 @@ class Document: NSDocument {
         self.sendRpcAsync("scroll", params: [first, last])
     }
 
+    /// handles the `update` RPC. This is called from a dedicated thread.
     func updateAsync(update: [String: AnyObject]) {
```

Comment:
I still feel like a reference to this function could still be in EditViewController, since it does modify the state of the EditView.

Reply:
There's a reason for this change, related to moving the `LineCache` into `Document`:

Basically with this change, we can have a situation where the first `update` arrives before the view controller is initialized. This wasn't possible before, because we blocked while waiting for the view_id to arrive, and so we could always finish setup before core sent `update`.

By moving the `LineCache` into the document, we can handle `update` even when there's no view controller; when the view controller actually gets initialized, the line cache will be up to date.